android/sdl_android/src/androidTest/java/com/smartdevicelink/transport/RSVTestCase.java

```diff
@@ -427,5 +429,35 @@ protected boolean isServiceRunning(Context context, ComponentName service){
 		assertTrue(rsvpPass.validate());
 	}
 
-	 
+	/**
+	 * Unit test for validateAsync.
+	 */
+	public void testValidateAsync() {
+		MultiplexTransportConfig config = new MultiplexTransportConfig(mContext, APP_ID, MultiplexTransportConfig.FLAG_MULTI_SECURITY_HIGH);
+		final RouterServiceValidator validator = new RouterServiceValidator(config);
+		final ConditionVariable cond = new ConditionVariable();
+		validator.validateAsync(new RouterServiceValidator.ValidationStatusCallback() {
+			@Override
+			public void onFinishedValidation(boolean valid, ComponentName name) {
+				Log.d(TAG, "onFinishedValidation: valid=" + valid + "; componentName=" + name);
+				assertFalse(valid); // expected valid = false for this (bogus) APP_ID..
+				cond.open();
+			}
+		});
+		cond.block();
+
+		// next, test for FLAG_MULTI_SECURITY_OFF
+		config = new MultiplexTransportConfig(mContext, APP_ID, MultiplexTransportConfig.FLAG_MULTI_SECURITY_OFF);
+		cond.close();
+		validator.validateAsync(new RouterServiceValidator.ValidationStatusCallback() {
```

Comment:
Shouldn't we pass the new config to the validator?  because now the new config that has `FLAG_MULTI_SECURITY_OFF` is not used at all.

Reply:
Oops, sorry, I should have created another final MultiplexTransportConfig instance, along with another final RouterServiceValidator instance... Thanks for pointing that out.